pkg/controller/jobs/rayjob/rayjob_controller.go

```diff
@@ -144,11 +144,16 @@ func (j *RayJob) PodSets() ([]kueue.PodSet, error) {
 	// submitter Job
 	if j.Spec.SubmissionMode == rayv1.K8sJobMode {
 		submitterJobPodSet := kueue.PodSet{
-			Name:  submitterJobPodSetName,
-			Count: 1,
+			Name:     submitterJobPodSetName,
+			Count:    1,
+			Template: *getSubmitterTemplate(j),
 		}
 
-		submitterJobPodSet.Template = *getSubmitterTemplate(j)
+		// Create the TopologyRequest for the Submitter Job PodSet, based on the annotations
+		// in rayJob.Spec.SubmitterPodTemplate, which can be specified by the user.
+		if jobframework.IsTopologyRequestFound(&submitterJobPodSet.Template.ObjectMeta) {
+			submitterJobPodSet.TopologyRequest = jobframework.PodSetTopologyRequest(&j.Spec.SubmitterPodTemplate.ObjectMeta, nil, nil, nil)
+		}
```

Comment:
```suggestion
   submitterJobPodSet.TopologyRequest = jobframework.PodSetTopologyRequest(&submitterJobPodSet.Template.ObjectMeta, nil, nil, nil)
```
why not just like this? The `PodSetTopologyRequest` function returns nil if there are not TAS annotations, so it should be safe to call without the extra "if". If this is the case then drop `IsTopologyRequestFound`